test/src/jupyterhub_config/custom_spawner_test.py

```diff
@@ -1 +1,83 @@
-from jupyterhub_config.custom_spawner import *
\ No newline at end of file
+import subprocess
+from subprocess import CalledProcessError
+from unittest.mock import patch, MagicMock
+
+import pytest
+
+from jupyterhub_config.custom_spawner import VirtualEnvSpawner
+
+
+@pytest.fixture
+def mock_spawner():
+    spawner = VirtualEnvSpawner()
+    spawner.log = MagicMock()  # Mock the logger
```

Comment:
Hmm, if the superclass logger is not at this location or has a different interface this won't catch it

Reply:
https://docs.pytest.org/en/7.1.x/how-to/logging.html#caplog-fixture might be useful, although I don't know how the logger in the superclass works

Reply:
But the tests don't check that it's working fine, that's my point. The logger could be at `logger` instead of `log` and the tests would pass

Reply:
I don't get what you mean here. The code is using `self.log` from parent class.

Reply:
test also checks every `mock_spawner.log.info` calls.

Reply:
My point is that the test isn't actually testing the logger. The code under test is calling `super.log.info`, but that's getting mocked, and then the output of the mock is checked. That means if there's a bug in the code and the correct call was `super.logger.log_info`, for example, the test wouldn't catch it.

Reply:
hmmm under what situation will that happen? I am more confused about `super.logger.log_info`. There is no `super.logger` involved.

Reply:
you mean something goes wrong with `self.log.info` function call?

Reply:
👍 using caplog to capture the logger output.

Reply:
:+1: